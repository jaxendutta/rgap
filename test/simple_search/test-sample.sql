-- Sample query for sp_search_institutes
SELECT 
  i.institute_id,
  i.name,
  COUNT(DISTINCT r.recipient_id) AS recipients_count,
  COUNT(DISTINCT rg.grant_id) AS grant_count,
  COALESCE(SUM(rg.agreement_value), 0) AS total_funding,
  MAX(rg.agreement_start_date) AS latest_grant_date,
  TRUE AS is_bookmarked  -- For testing, we simply return TRUE
FROM Institute i
LEFT JOIN Recipient r ON i.institute_id = r.institute_id
LEFT JOIN ResearchGrant rg ON r.recipient_id = rg.recipient_id
WHERE i.name LIKE '%Test%'
   OR i.city LIKE '%Test%'
   OR i.province LIKE '%Test%'
   OR i.country LIKE '%Test%'
GROUP BY i.institute_id
ORDER BY total_funding DESC
LIMIT 10 OFFSET 0;
