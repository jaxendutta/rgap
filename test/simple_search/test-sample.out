+--------------+--------------------------------+------------------+-------------+---------------+-------------------+---------------+
| institute_id | name                           | recipients_count | grant_count | total_funding | latest_grant_date | is_bookmarked |
+--------------+--------------------------------+------------------+-------------+---------------+-------------------+---------------+
|          319 | Testlabs International Ltd     |                2 |           4 |      18000.00 | 2018-09-04        |             1 |
|          375 | CellScale Biomaterials Testing |                4 |           4 |      18000.00 | 2018-05-07        |             1 |
|         2559 | Caribou Contests Inc           |                3 |           3 |      13500.00 | 2017-09-20        |             1 |
|          213 | Caribou Contests Inc           |                2 |           2 |       9000.00 | 2018-01-01        |             1 |
|         3128 | CellScale Biomaterials Testing |                2 |           2 |       8718.75 | 2018-03-07        |             1 |
|          823 | Nova Medical Testing Inc       |                1 |           1 |       4500.00 | 2018-05-01        |             1 |
|         3245 | Testlabs International Ltd     |                1 |           1 |       4500.00 | 2017-11-08        |             1 |
+--------------+--------------------------------+------------------+-------------+---------------+-------------------+---------------+
7 rows in set (0.01 sec)
